ui/imports/shared/popups/AccountsModalHeader.qml

```diff
@@ -1,83 +1,89 @@
+    property string chainShortNames
+    property int selectedIndex: 0
 
-    repeater.objectName: "accountsListFloatingHeader"
+    QtObject {
+        id: d
+        function getTextColorForWhite(color) {
+            // The grey is kept for backwards compatibility for accounts already created with grey background
+            return color === StatusColors.colors['grey'] || color === StatusColors.colors['white'] ? Theme.palette.black : Theme.palette.white
+        }
+    }
 
-    signal updatedSelectedAccount(var account)
+    control.padding: 0
+    control.spacing: 0
+    control.leftPadding: 8
+    control.rightPadding: 8
+    control.topPadding: 10
 
-    delegate: Rectangle {
-        width: button.width
-        height: button.height
+    control.popup.width: 430
+    control.indicator: null
+
+    control.background: Rectangle {
+        width: contentItem.childrenRect.width + control.leftPadding + control.rightPadding
+        height: 32
         radius: 8
-        color: Theme.palette.baseColor3
-        StatusButton {
-            id: button
-            size: StatusBaseButton.Size.Tiny
-            implicitHeight: 32
-            leftPadding: 4
-            text: name
-            objectName: name
-            asset.emoji: !!emoji ? emoji: ""
-            icon.name: !emoji ? "filled-account": ""
-            normalColor: "transparent"
-            hoverColor: Theme.palette.statusFloatingButtonHighlight
-            highlighted: index === root.currentIndex
-            onClicked: {
-                changeSelectedAccount(model)
-                root.currentIndex = index
-            }
-            Component.onCompleted: {
-                // on model reset, set the selected account to the one that was previously selected
-                if(root.selectedAccount === null) {
-                    if(root.currentIndex === index) {
-                        changeSelectedAccount(model)
-                    }
-                }
-                else {
-                    // if the selectedAccount is watch only then select 0th item
-                    if(index === 0 && !!root.selectedAccount && root.selectedAccount.walletType === Constants.watchWalletType) {
-                        changeSelectedAccount(model)
-                        root.currentIndex = index
-                    }
-                }
-            }
+        color: !!selectedAccount ? sensor.containsMouse ?
+                                       Theme.palette.walletAccountColors.getHoveredColor(selectedAccount.color) :
+                                       selectedAccount.color : "transparent"
+        MouseArea {
+            id: sensor
+            anchors.fill: parent
+            cursorShape: Qt.PointingHandCursor
+            acceptedButtons: Qt.NoButton
+            hoverEnabled: enabled
+            enabled: root.enabled
+        }
+    }
+
+    contentItem: Row {
+        anchors.verticalCenter: parent.verticalCenter
+        width: childrenRect.width
```

Comment:
`anchors` or `width` should not be needed at all, the Control will resize it to fit

Reply:
Here it should work without tho

Reply:
So here, If I do not add this I get binding width error, because the background depends on the contentItem.childrenrect.width which determines width of the contentItem